src/titiler/core/titiler/core/algorithm/dem.py

```diff
@@ -161,6 +168,7 @@ class Terrarium(BaseAlgorithm):
 
     title: str = "Terrarium"
     description: str = "Encode DEM into RGB (Mapzen Terrarium)."
+    nodata_height: Optional[float] = Field(None, ge=-99999.0, le=99999.0)
```

Comment:
The issue is when you don't know much about the data (e.g no idea about the `min`) 

we could automagically set the `nodata_height` to the `min` from the raster statistic (if present)

Reply:
or even one step further 

```python
        if self.nodata_height is not None:
            data[img.array.mask[0]] = (self.nodata_height - self.baseval) / self.interval
        elif img.dataset_statistics:
            # Check image statistics 
            stats = img.dataset_statistics[0]
            data[img.array.mask[0]] = (stats[0] - self.baseval) / self.interval
        else:
            # Get min value from the image (tile)
            data[img.array.mask[0]] = (img.array[0].min() - self.baseval) / self.interval
```

Reply:
These defaults do look great indeed in case there's no user-provided nodata_height